### PR TITLE
Adding golang application for asset-transfer-basic sample.

### PR DESCRIPTION
**Pull request by DereckLuo**
Signed-off-by: Chongxin Luo <Chongxin.Luo@ibm.com>

**Description**

* Adding Golang application for `asset-transfer-basic` sample.

### CODE REVIEW

File: asset-transfer-basic/application-go/assetTransfer.go [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+
+	fmt.Println("connect to channel: mychannel")
+	network, err := gw.GetNetwork("mychannel")
+	if err != nil {
+		fmt.Printf("Failed to get network: %s\n", err)
+		os.Exit(1)
+	}
+
+	fmt.Println("Get smart contract")
+	contract := network.GetContract("fabcar")
```
Comment (sijocherian) [inline asset-transfer-basic/application-go/assetTransfer.go:64]: not fabcar. Better to declare a const at top, for the `ContractName`

File: asset-transfer-basic/application-go/assetTransfer.go [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+
+	fmt.Println("Evaluate transaction --- getAllAssets")
+	result, err := contract.EvaluateTransaction("getAllAssets")
+	if err != nil {
+		fmt.Printf("Failed to evaluate transaction: %s\n", err)
+		os.Exit(1)
+	}
+	fmt.Println(string(result))
+
+	fmt.Println("Submit transaction --- createAsset")
```
Comment (sijocherian) [inline asset-transfer-basic/application-go/assetTransfer.go:74]: Add detail in the println. for e.g. ` --- createAsset asset13`

File: asset-transfer-basic/application-go/assetTransfer.go [hunk 3]
```diff
@@ -0,0 +1,151 @@
...
+	ccpPath := filepath.Join(
+		"..",
+		"..",
+		"test-network",
+		"organizations",
+		"peerOrganizations",
+		"org1.example.com",
+		"connection-org1.yaml",
+	)
+
```
Comment (sijocherian) [inline asset-transfer-basic/application-go/assetTransfer.go:45]: for this sample to work on one host, isn't this env variable needed?
`export DISCOVERY_AS_LOCALHOST=TRUE`

can we set all the ENV vars needed at beginning of each go application sample?

File: asset-transfer-basic/application-go/assetTransfer.go [hunk 4]
```diff
@@ -0,0 +1,151 @@
...
+	fmt.Println("Evaluate transacation --- getAsset")
+	result, err = contract.EvaluateTransaction("getAsset", "asset4")
+	if err != nil {
+		fmt.Printf("Failed to evaluate transaction: %s\n", err)
+		os.Exit(1)
+	}
+	fmt.Println(string(result))
+
+	fmt.Println("Submit transaction --- transferAsset")
```
Comment (sijocherian) [inline asset-transfer-basic/application-go/assetTransfer.go:90]: println. for e.g. `--- transferAsset with id asset1 to new owner Tom`

File: asset-transfer-basic/application-go/assetTransfer.go [hunk 5]
```diff
@@ -0,0 +1,151 @@
...
+	_, err = contract.SubmitTransaction("transferAsset", "asset1", "Tom")
+	if err != nil {
+		fmt.Printf("Failed to submit transaction: %s\n", err)
+		os.Exit(1)
+	}
+
+	fmt.Println("Evaluate transaction --- getAsset")
```
Comment (sijocherian) [inline asset-transfer-basic/application-go/assetTransfer.go:97]: getAsset `asset1`

File: asset-transfer-basic/application-go/assetTransfer.go [hunk 6]
```diff
@@ -0,0 +1,151 @@
...
+	result, err = contract.EvaluateTransaction("getAsset", "asset1")
```
Comment (sijocherian) [inline asset-transfer-basic/application-go/assetTransfer.go:98]: actual chaincode txn is called `readAsset` not `getAsset`

File: asset-transfer-basic/application-go/assetTransfer.go (hunk 3 above)
Comment (sijocherian) [reply asset-transfer-basic/application-go/assetTransfer.go:45]: note #232 PR, added instruction to set DISCOVERY_AS_LOCALHOST

File: asset-transfer-basic/application-go/assetTransfer.go (hunk 1 above)
Comment (DereckLuo) [reply asset-transfer-basic/application-go/assetTransfer.go]: done.

File: asset-transfer-basic/application-go/assetTransfer.go (hunk 2 above)
Comment (DereckLuo) [reply asset-transfer-basic/application-go/assetTransfer.go]: this was used as debug statement, has been removed from the application. 
same applies for rest prints.

File: asset-transfer-basic/application-go/assetTransfer.go (hunk 6 above)
Comment (DereckLuo) [reply asset-transfer-basic/application-go/assetTransfer.go]: done.

File: asset-transfer-basic/application-go/assetTransfer.go (hunk 3 above)
Comment (DereckLuo) [reply asset-transfer-basic/application-go/assetTransfer.go:47]: thanks for pointing it out. this was the cause of fabcar golang application not runnable. 
I added `os.Setenv("DISCOVERY_AS_LOCALHOST", "true")` to the beginning of `assetTransfer.go`, but still want to test it with new deployment script to confirm.

**Comment by DereckLuo** [comment]
@sijocherian Thanks for the review, changes has been applied accordingly, and this PR is ready to be reviewed.

File: asset-transfer-basic/application-go/assetTransfer.go [hunk 7]
```diff
@@ -0,0 +1,146 @@
...
+
+	network, err := gw.GetNetwork("mychannel")
+	if err != nil {
+		fmt.Printf("failed to get network: %v\n", err)
+		os.Exit(1)
+	}
+
+	contract := network.GetContract("basic")
+
+	result, err := contract.EvaluateTransaction("GetAllAssets")
```
Comment (lehors) [inline asset-transfer-basic/application-go/assetTransfer.go:66]: InitLedger needs to be called first.

File: asset-transfer-basic/application-go/assetTransfer.go [hunk 8]
```diff
@@ -0,0 +1,146 @@
...
+	if err != nil {
+		fmt.Printf("failed to evaluate transaction: %v\n", err)
+		os.Exit(1)
+	}
+	fmt.Println(string(result))
+
+	result, err = contract.SubmitTransaction("CreateAsset", "asset13", "yellow", "Tom", "5", "1300")
```
Comment (lehors) [inline asset-transfer-basic/application-go/assetTransfer.go:73]: The order of the parameters is wrong, you need to swap owner and size, Tom and 5.

**Comment by lehors** [comment]
> Putting a change-request on this to block its merge until I finish CI and linters. In the meantime a recommend you at least run your code through ESLint and the Go Linter as CI will block anything caught by these linters.
> 
> I'm pretty packed tomorrow, but hoping to finish CI work on Friday now that we got Typescript removed.

@btl5037 what's the status on this?

**Comment by btl5037** [comment]
> > Putting a change-request on this to block its merge until I finish CI and linters. In the meantime a recommend you at least run your code through ESLint and the Go Linter as CI will block anything caught by these linters.
> > I'm pretty packed tomorrow, but hoping to finish CI work on Friday now that we got Typescript removed.
> 
> @btl5037 what's the status on this?

This work is done, Dereck, do you want to add this to CI, or is it ok if I push a change on top of your here for you?

**Comment by DereckLuo** [comment]
> > > Putting a change-request on this to block its merge until I finish CI and linters. In the meantime a recommend you at least run your code through ESLint and the Go Linter as CI will block anything caught by these linters.
> > > I'm pretty packed tomorrow, but hoping to finish CI work on Friday now that we got Typescript removed.
> > 
> > 
> > @btl5037 what's the status on this?
> 
> This work is done, Dereck, do you want to add this to CI, or is it ok if I push a change on top of your here for you?

@btl5037 not sure how to add to CI, If you could add the change on top that will be great.

**Pull request merged**